runtime/metricsview/executor_summary.go

```diff
@@ -0,0 +1,250 @@
+			timeDimensions = append(timeDimensions, dim)
+		} else {
+			dimensions = append(dimensions, dim)
+		}
+	}
+
+	// Get the default time range for the metrics view
+	defaultTimeRange, err := e.Timestamps(ctx, timeDimension)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get default time range: %w", err)
+	}
+
+	timeDimensionSummaries := make([]DimensionSummary, 0, len(timeDimensions))
+
+	// Validate time range
+	if defaultTimeRange.Max.IsZero() {
+		return &SummaryResult{
+			Dimensions: timeDimensionSummaries,
+			TimeRange:  defaultTimeRange,
+		}, nil
+	}
+
+	for _, dim := range timeDimensions {
+		timeRange, err := e.Timestamps(ctx, dim.Name)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get time range for dimension %s: %w", dim.Name, err)
+		}
+
+		var dataType string
+		if dim.DataType != nil {
+			dataType = dim.DataType.Code.String()
+		}
+
+		timeDimensionSummaries = append(timeDimensionSummaries, DimensionSummary{
+			Name:      dim.Name,
+			TimeRange: timeRange,
+			DataType:  dataType,
+		})
+	}
+
+	// Create summaries for normal dimensions
+	if len(dimensions) == 0 {
+		return &SummaryResult{
+			Dimensions: timeDimensionSummaries,
+			TimeRange:  defaultTimeRange,
+		}, nil
+	}
+
+	// Build and execute the summary query
+	selectClauses := make([]string, 0, len(dimensions)*4)
+	var failedDimensions []string
+
+	for _, dim := range dimensions {
+		dimName := dim.Name
+		expr, err := e.olap.Dialect().MetricsViewDimensionExpression(dim)
+		if err != nil {
+			failedDimensions = append(failedDimensions, dimName)
+			continue // Skip this dimension instead of failing entirely
+		}
+
+		dialect := e.olap.Dialect()
+
+		// Add select clauses for min, max, has_nulls, and sample value
+		selectClauses = append(selectClauses,
+			fmt.Sprintf("MIN(%s) AS %s", expr, dialect.EscapeIdentifier(dimName+"__min")),
+			fmt.Sprintf("MAX(%s) AS %s", expr, dialect.EscapeIdentifier(dimName+"__max")),
+			fmt.Sprintf("COUNT(*) - COUNT(%s) > 0 AS %s", expr, dialect.EscapeIdentifier(dimName+"__has_nulls")),
+			fmt.Sprintf("ANY_VALUE(%s) AS %s", expr, dialect.EscapeIdentifier(dimName+"__sample")),
+		)
+	}
+
+	if len(selectClauses) == 0 {
+		if len(failedDimensions) > 0 {
+			return nil, fmt.Errorf("failed to get expressions for all dimensions: %v", failedDimensions)
+		}
+		return nil, fmt.Errorf("no dimensions to summarize")
+	}
+
+	timeDimExpr := e.olap.Dialect().EscapeIdentifier(e.metricsView.TimeDimension)
+	if e.metricsView.TimeDimension != "" {
+		for _, dim := range e.metricsView.Dimensions {
+			if dim.Name == e.metricsView.TimeDimension {
+				expr, err := e.olap.Dialect().MetricsViewDimensionExpression(dim)
+				if err == nil {
+					timeDimExpr = expr
+				}
+				break
+			}
+		}
+	}
+
+	// Use a recent time window with proper timestamp formatting
+	dimensionTimeRange := defaultTimeRange.Max.Add(-DefaultSampleTimeWindow)
+	whereClause := fmt.Sprintf("WHERE %s >= '%s'", timeDimExpr, dimensionTimeRange.Format(time.RFC3339))
```

Comment:
The metrics view may not have a time dimension. So make these checks conditional on whether a time range/expression was found.